CS3230Project/Model/Users/Nurses/Nurse.cs

```diff
@@ -0,0 +1,240 @@
+            {
+                throw new ArgumentException(NurseErrorMessages.LastNameCannotBeNull);
+            }
+            if (lastName.Trim().Length == 0)
+            {
+                throw new ArgumentException(NurseErrorMessages.LastNameCannotBeEmpty);
+            }
+            if (dateOfBirth < new DateTime(1900, 1, 1))
+            {
+                throw new ArgumentException(NurseErrorMessages.DateOfBirthCannotBeBefore1900);
+            }
+            if (dateOfBirth > DateTime.Now)
+            {
+                throw new ArgumentException(NurseErrorMessages.DateOfBirthCannotBeInTheFuture);
+            }
+            if (gender == null)
+            {
+                throw new ArgumentException(NurseErrorMessages.GenderCannotBeNull);
+            }
+            if (gender.Trim().Length == 0)
+            {
+                throw new ArgumentException(NurseErrorMessages.GenderCannotBeEmpty);
+            }
+            if (gender.Length > UserSettings.GenderMaximumLength)
+            {
+                throw new ArgumentException(NurseErrorMessages.GenderIsTooLong);
+            }
+            if (phoneNumber == null)
+            {
+                throw new ArgumentException(NurseErrorMessages.PhoneNumberCannotBeNull);
+            }
+            if (phoneNumber.Trim().Length == 0)
+            {
+                throw new ArgumentException(NurseErrorMessages.PhoneNumberCannotBeEmpty);
+            }
+            if (!DataValidator.IsValidPhoneNumberFormat(phoneNumber))
+            {
+                throw new ArgumentException(NurseErrorMessages.InvalidPhoneNumberFormat);
+            }
+            if (addressOne == null)
+            {
+                throw new ArgumentException(NurseErrorMessages.AddressOneCannotBeNull);
+            }
+            if (addressOne.Trim().Length == 0)
+            {
+                throw new ArgumentException(NurseErrorMessages.AddressOneCannotBeEmpty);
+            }
+            if (addressOne.Length > UserSettings.AddressComponentMaximumLength)
+            {
+                throw new ArgumentException(NurseErrorMessages.AddressOneIsTooLong);
+            }
+            if (city == null)
+            {
+                throw new ArgumentException(NurseErrorMessages.CityCannotBeNull);
+            }
+            if (city.Trim().Length == 0)
+            {
+                throw new ArgumentException(NurseErrorMessages.CityCannotBeEmpty);
+            }
+            if (city.Length > UserSettings.AddressComponentMaximumLength)
+            {
+                throw new ArgumentException(NurseErrorMessages.CityIsTooLong);
+            }
+            if (state == null)
+            {
+                throw new ArgumentException(NurseErrorMessages.StateCannotBeNull);
+            }
+            if (state.Trim().Length == 0)
+            {
+                throw new ArgumentException(NurseErrorMessages.StateCannotBeEmpty);
+            }
+            if (state.Length > 50)
```

Comment:
There is a constant in the settings class for the maximum lengths

Reply:
Will be fixed.

Reply:
Fixed.